test/src/Test/Finance/Asset/Account.daml

```diff
@@ -0,0 +1,97 @@
+module Finance.Asset.Account where
+
+import Daml.Script
+import DA.Assert ((===))
+import DA.Finance.Asset.Account qualified as Account
+import DA.Finance.Asset
+import DA.Finance.Types
+import DA.Set
+import DA.Optional
+
+accountRuleLockingTest : Script ()
+accountRuleLockingTest = do
+  now <- getTime
+
+  -- Create our parties
+  [alice, bob, bank, exchange, issuer, auditor] <- forA ["Alice", "Bob", "Bank", "Exchange", "Issuer", "Auditor"] \party -> allocatePartyWithHint party $ PartyIdHint party
+
+  -- Alice creates an account at the bank -> Gather Alice's and the Bank signatures and create account to simulate a real life use case
+  let accountId = \party -> Id with signatories = fromList [ bank, party ]; label = partyToText party <> "@" <> partyToText bank; version = 0
+      account = \party -> Account with provider = bank; owner = party; id = accountId party
+      assetId = Id with signatories = fromList [ bank, issuer]; label = "btc"; version = 0
+      asset = Asset with id = assetId; quantity = 1.0
+
+  -- Create an account for Alice and Bob
+  aliceAccountRuleCid <- submitMulti [ alice, bank ] [] $ createCmd Account.AccountRule with account = account alice; nominees = fromList [ exchange, bank ]; observers = empty
+  bobAccountRuleCid   <- submitMulti [ bob, bank ]   [] $ createCmd Account.AccountRule with account = account bob;   nominees = fromList [ exchange, bank ]; observers = empty
+
+  -- Alice and the Bank both attempt on their own to Credit the account
+  alice `submitMustFail` exerciseCmd aliceAccountRuleCid Account.Credit with asset; nominee = alice  -- Fails as Alice is not an nominee
+  bank  `submitMustFail` exerciseCmd aliceAccountRuleCid Account.Credit with asset; nominee = bank   -- Fails due to missing auth from Alice
+
+  -- Credit the account with both the signature of alice and the bank
+  depositCid <- submitMulti [ alice, bank ] [] $ exerciseCmd aliceAccountRuleCid Account.Credit with asset; nominee = bank
```

Comment:
does this work because the bank happens to also be in the signatories? so it's not the fact that the bank is a nominee, but rather that it is the account provider, and therefore, together with the owner, can create a deposit out of thin air?

What I'm worried here is that we give a nominee the power to create a deposit in an account that's backed (ie. provided) by a different party (which would be wrong). but i don't think that's the case, since we need the `account.id.signatories` to be able to create the asset deposit.

Reply:
This works because in the Account.Credit choice, if you have the authority to exercise the choice and all the assertions pass - you have all the signatures available as they have signed the rule template.  For example, this statement passes when the `exchange` is a nominee on an account when the signatory of the account is both `alice` and the `bank` - 

> depositCid <- submitMulti [ alice, exchange ] [] $ exerciseCmd aliceAccountRuleCid Account.Credit with asset; nominee = exchange  

We need to decide if my implementation has given the nominee too much power - My thought process for this is that the signatories must sign for the `nominees` that are contained in that set, do we need to have their signature each time a `nominee` wishes to create a deposit ?